BioCommons/src/test/java/uk/ac/ebi/biostd/commons/files/MagicFolderUtilTest.java

```diff
@@ -1,91 +1,96 @@
+    private static String symlinksPath;
 
     private MagicFolderUtil testInstance;
 
-    @Before
-    public void setup() {
-        when(mockProperties.get(USER_GROUP_DIR_PROP_NAME)).thenReturn(TEST_FOLDER.getRoot().getAbsolutePath());
-        testInstance = new MagicFolderUtil(mockProperties);
+    @BeforeAll
+    static void beforeAll(@TempDir Path tempDir) {
+        testFolderPath = tempDir.toAbsolutePath().toString();
+    }
 
-        rootPath = TEST_FOLDER.getRoot().getAbsolutePath();
+    @BeforeEach
+    void setup() {
+        symlinksPath = testFolderPath + "/symlinks";
+        testInstance = new MagicFolderUtil(testFolderPath, symlinksPath);
     }
 
     @Test
-    public void createUserMagicFolder() {
+    void createUserMagicFolder() {
         testInstance.createUserMagicFolder(50L, "abc-123");
 
-        File parentFolder = new File(rootPath, "ab");
+        File parentFolder = new File(testFolderPath, "ab");
         assertThat(parentFolder).exists();
 
         File magicFolder = new File(parentFolder, "c-123-a50");
         assertThat(magicFolder).exists();
     }
 
     @Test
-    public void createGroupMagicFolder() {
+    void createGroupMagicFolder() {
         testInstance.createGroupMagicFolder(40L, "abc-123");
 
-        File parentFolder = new File(rootPath, "ab");
+        File parentFolder = new File(testFolderPath, "ab");
         assertThat(parentFolder).exists();
 
         File magicFolder = new File(parentFolder, "c-123-b40");
         assertThat(magicFolder).exists();
     }
 
     @Test
-    public void getUserMagicFolderPath() {
+    void getUserMagicFolderPath() {
         Path userMagicFolderPath = testInstance.getUserMagicFolderPath(TEST_ID, TEST_SECRET);
         assertThat(userMagicFolderPath).isEqualTo(getExpectedPath(MagicFolderUtil.USER_FOLDER_PREFIX));
     }
 
     @Test
-    public void getGroupMagicFolderPath() {
+    void getGroupMagicFolderPath() {
         Path userMagicFolderPath = testInstance.getGroupMagicFolderPath(TEST_ID, TEST_SECRET);
         assertThat(userMagicFolderPath).isEqualTo(getExpectedPath(MagicFolderUtil.GROUP_FOLDER_PREFIX));
     }
 
     @Test
-    public void createMagicFolderException() {
-        when(mockProperties.get(USER_GROUP_DIR_PROP_NAME)).thenReturn("/folder");
-        MagicFolderUtil faultyTestInstance = new MagicFolderUtil(mockProperties);
+    void createMagicFolderException() {
+        MagicFolderUtil faultyTestInstance = new MagicFolderUtil("/folder", "symlinks");
 
         assertThatExceptionOfType(NoSuchFileException.class).isThrownBy(
                 () -> faultyTestInstance.createUserMagicFolder(TEST_ID, TEST_SECRET));
     }
 
+    @Test
+    void createUserSymlink() throws Exception {
+        Path source = Paths.get(testFolderPath + "/directory");
+        source.toFile().mkdir();
+
+        testInstance.createUserSymlink(source, "user_email");
+        assertThat(Paths.get(symlinksPath + "/u/user_email").toRealPath()).isEqualTo(source);
```

Comment:
Also, I think we should have an agreement about using constants vs direct values in tests. Both have up/downsides but we should choose, not mix :smile: